index.html

```diff
@@ -2599,6 +2608,124 @@ <h3>
         </p>
       </section>
     </section>
+    <section>
+      <h2>
+        API Definition
+      </h2>
+      <section data-dfn-for="Navigator">
+        <h3>
+          Extensions to the `Navigator` interface
+        </h3><!-- todo: parameters -->
+        <!-- todo: options bag -->
+        <!-- todo: return value -->
+        <pre class="idl">
+          partial interface Navigator {
+              [SecureContext, NewObject] Promise&lt;undefined&gt; install();
+          };
+        </pre>
+        <h3>
+          Internal Slots
+        </h3>
+        <p>
+          This API adds the following internal slot to the {{Navigator}}
+          interface.
+        </p>
+        <dl>
+          <dt>
+            {{Promise}}? <dfn>[[\installPromise]]</dfn>
+          </dt>
+          <dd>
+            The [=this=].{{Navigator/[[installPromise]]}} is a promise that
+            represents a user's current intent to install an application. It is
+            initialized to `null`.
+          </dd>
+        </dl>
+        <h3>
+          <dfn>install()</dfn> method
+        </h3>
+        <p>
+          When the {{Navigator/install()}} method is called, run the following
+          steps to [=install=] the website.
+        </p>
+        <ol class="algorithm">
+          <li>Let |global:Window| be [=this=]'s [=relevant global object=].
+          </li>
+          <li>Let |document:Document| be |global|'s [=associated Document=].
+          </li>
+          <li>If |document| is not [=Document/fully active=], return [=a
+          promise rejected with=] an {{"InvalidStateError"}} {{DOMException}}.
+          </li><!-- todo: permission policy? -->
```

Comment:
Instead of a permission policy, this should throw with a NotAllowedError if the browsing context is not the top-level traversable.